src/R/Components/Impl/Repl/Commands/SourceRScriptCommand.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.R.Components.Controller;
+using Microsoft.R.Components.Extensions;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.R.Package.Repl.Commands {
+    public sealed class SourceRScriptCommand : IMenuCommand {
+        private readonly IRInteractiveWorkflow _interactiveWorkflow;
+        private readonly IActiveWpfTextViewTracker _activeTextViewTracker;
+        private readonly bool _echo;
+
+        public SourceRScriptCommand(
+            IRInteractiveWorkflow interactiveWorkflow,
+            IActiveWpfTextViewTracker activeTextViewTracker,
+            bool echo
+        ) {
+            _interactiveWorkflow = interactiveWorkflow;
+            _activeTextViewTracker = activeTextViewTracker;
+            _echo = echo;
+        }
+
+        public CommandStatus Status {
+            get {
+                var status = CommandStatus.Supported;
+                if (_interactiveWorkflow.ActiveWindow == null) {
+                    status |= CommandStatus.Invisible;
+                } else if (!string.IsNullOrEmpty(GetFilePath())) {
+                    status |= CommandStatus.Enabled;
+                }
+                return status;
+            }
+        }
+
+        public CommandResult Invoke(object inputArg, ref object outputArg) {
```

Comment:
It's `ref` because that's what `ICommandTarget` uses. I'm assuming this is originally to make the implementations shorter, since there's no reason why it would need to read from it.

`CommandResult` is indeed currently unused, but if we ever want to support command chaining for these, it may come in handy.